config/gasSchedule_test.go

```diff
@@ -84,22 +84,82 @@ func TestDecode_ZeroGasCostError(t *testing.T) {
 func Test_getSignedCoefficient(t *testing.T) {
 	gasScheduleMap := MakeGasMap(1, 1)
 
-	A := uint64(687)
-	B := uint64(30483)
-	C := uint64(15883)
+	A := uint64(688)
+	B := uint64(31858)
+	C := uint64(15287)
 
 	gasMap := make(map[string]uint64)
 	gasMap["QuadraticCoefficient"] = A
 	gasMap["SignOfQuadratic"] = 0
 	gasMap["LinearCoefficient"] = B
 	gasMap["SignOfLinear"] = 0
 	gasMap["ConstantCoefficient"] = C
-	gasMap["SignOfConstant"] = 1
+	gasMap["SignOfConstant"] = 0
 	gasScheduleMap["DynamicStorageLoad"] = gasMap
 
 	gasCost, err := CreateGasConfig(gasScheduleMap)
 	assert.Nil(t, err)
 	assert.Equal(t, int64(A), gasCost.DynamicStorageLoad.Quadratic)
 	assert.Equal(t, int64(B), gasCost.DynamicStorageLoad.Linear)
-	assert.Equal(t, int64(C)*-1, gasCost.DynamicStorageLoad.Constant)
+	assert.Equal(t, int64(C), gasCost.DynamicStorageLoad.Constant)
+}
+
+func Test_isDynamicGasComputationFuncCorrectlyDefined(t *testing.T) {
+	t.Parallel()
+
+	t.Run("invalid inflection point", func(t *testing.T) {
+		t.Parallel()
+
+		params := &DynamicStorageLoadCostCoefficients{
+			Quadratic:  5,
+			Linear:     -5,
+			Constant:   1,
+			MinGasCost: 0,
+		}
+
+		ok := isDynamicGasComputationFuncCorrectlyDefined(params)
+		assert.False(t, ok)
+	})
+
+	t.Run("concave func", func(t *testing.T) {
+		t.Parallel()
+
+		params := &DynamicStorageLoadCostCoefficients{
+			Quadratic:  -5,
+			Linear:     -5,
+			Constant:   1,
+			MinGasCost: 0,
+		}
+
+		ok := isDynamicGasComputationFuncCorrectlyDefined(params)
+		assert.False(t, ok)
+	})
+
+	t.Run("constant parameter is positive", func(t *testing.T) {
+		t.Parallel()
+
+		params := &DynamicStorageLoadCostCoefficients{
+			Quadratic:  5,
+			Linear:     5,
+			Constant:   -1,
+			MinGasCost: 0,
+		}
+
+		ok := isDynamicGasComputationFuncCorrectlyDefined(params)
+		assert.False(t, ok)
+	})
+
+	t.Run("ok params", func(t *testing.T) {
+		t.Parallel()
+
+		params := &DynamicStorageLoadCostCoefficients{
+			Quadratic:  688,
+			Linear:     31858,
+			Constant:   15287,
+			MinGasCost: 0,
+		}
+
+		ok := isDynamicGasComputationFuncCorrectlyDefined(params)
+		assert.True(t, ok)
```

Comment:
maybe add also the test "quadratic = 5, linear = 5, constant = 1", should work

Reply:
Added.